java-checks/src/main/java/org/sonar/java/checks/spring/StatusCodesOnResponseCheck.java

```diff
@@ -0,0 +1,222 @@
+      return;
+    }
+
+    MethodInvocationVisitor methodInvocationVisitor = new MethodInvocationVisitor();
+    classTree.accept(methodInvocationVisitor);
+  }
+
+  private class MethodInvocationVisitor extends BaseTreeVisitor {
+
+    MethodMatchers statusMethodMatchers = MethodMatchers.create()
+      .ofTypes(RESPONSE_ENTITY)
+      .names("status")
+      .addParametersMatcher("org.springframework.http.HttpStatus")
+      .build();
+
+    MethodMatchers okMethodMatchers = MethodMatchers.create()
+      .ofTypes(RESPONSE_ENTITY)
+      .names("ok", "created", "accepted", "noContent")
+      .withAnyParameters()
+      .build();
+
+    MethodMatchers errorMethodsMatcher = MethodMatchers.create()
+      .ofTypes(RESPONSE_ENTITY)
+      .names("badRequest", "notFound", "unprocessableEntity")
+      .addWithoutParametersMatcher()
+      .build();
+
+    @Override
+    public void visitMethodInvocation(MethodInvocationTree methodInvocationTree) {
+
+      if (statusMethodMatchers.matches(methodInvocationTree)) {
+        checkTryCatch(methodInvocationTree);
+      }
+
+      if (okMethodMatchers.matches(methodInvocationTree)) {
+        Tree catchParent = checkCatch(methodInvocationTree, false);
+        if (catchParent == null) {
+          checkTry(methodInvocationTree, true);
+        }
+      }
+
+      if (errorMethodsMatcher.matches(methodInvocationTree)) {
+        Tree catchParent = checkCatch(methodInvocationTree, true);
+        if (catchParent == null) {
+          checkTry(methodInvocationTree, false);
+        }
+      }
+
+      super.visitMethodInvocation(methodInvocationTree);
+    }
+
+    private void checkTryCatch(MethodInvocationTree methodInvocationTree) {
+      Tree catchParent = ExpressionUtils.getParentOfType(methodInvocationTree, Tree.Kind.CATCH);
+      boolean isError = isCodeInList(methodInvocationTree, errorCodes);
+
+      if (catchParent != null && !isError) {
+        reportIssue(methodInvocationTree, ERROR_ISSUE_MESSAGE);
+      }
+
+      if (catchParent == null) {
+        Tree tryParent = ExpressionUtils.getParentOfType(methodInvocationTree, Tree.Kind.TRY_STATEMENT);
+        boolean isOk = isCodeInList(methodInvocationTree, okCodes);
+
+        if (tryParent != null && !isOk) {
+          reportIssue(methodInvocationTree, OK_ISSUE_MESSAGE);
+        }
+      }
+    }
+
+    private boolean isCodeInList(MethodInvocationTree methodInvocationTree, List<String> codes) {
+      for (ExpressionTree arg : methodInvocationTree.arguments()) {
+        if (arg.is(Tree.Kind.MEMBER_SELECT)) {
+          MemberSelectExpressionTree memberSelectExpressionTree = (MemberSelectExpressionTree) arg;
+          return codes.contains(memberSelectExpressionTree.identifier().name());
+        } else if (arg.is(Tree.Kind.IDENTIFIER)) {
+          IdentifierTree identifierTree = (IdentifierTree) arg;
+          return codes.contains(identifierTree.name());
+        }
+      }
+      return true;
```

Comment:
My understanding is that the methods invoked here take up to 1 argument (since we systematically return after testing the first argument). Can we replace this `for` loop with a single `if` statement?